SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -299,114 +162,118 @@ void USpatialWorkerConnection::ConnectToLocator()
+	return NextRequestId++;
 }
 
 Worker_RequestId USpatialWorkerConnection::SendDeleteEntityRequest(Worker_EntityId EntityId)
 {
-	return Worker_Connection_SendDeleteEntityRequest(WorkerConnection, EntityId, nullptr);
+	QueueOutgoingMessage<FDeleteEntityRequest>(EntityId);
+	return NextRequestId++;
 }
 
 void USpatialWorkerConnection::SendComponentUpdate(Worker_EntityId EntityId, const Worker_ComponentUpdate* ComponentUpdate)
 {
-	Worker_Alpha_UpdateParameters UpdateParameters{};
-	UpdateParameters.loopback = false;
-	Worker_Alpha_Connection_SendComponentUpdate(WorkerConnection, EntityId, ComponentUpdate, &UpdateParameters);
+	QueueOutgoingMessage<FComponentUpdate>(EntityId, *ComponentUpdate);
 }
 
 Worker_RequestId USpatialWorkerConnection::SendCommandRequest(Worker_EntityId EntityId, const Worker_CommandRequest* Request, uint32_t CommandId)
 {
-	Worker_CommandParameters CommandParams{};
-	return Worker_Connection_SendCommandRequest(WorkerConnection, EntityId, Request, CommandId, nullptr, &CommandParams);
+	QueueOutgoingMessage<FCommandRequest>(EntityId, *Request, CommandId);
+	return NextRequestId++;
 }
 
 void USpatialWorkerConnection::SendCommandResponse(Worker_RequestId RequestId, const Worker_CommandResponse* Response)
 {
-	Worker_Connection_SendCommandResponse(WorkerConnection, RequestId, Response);
+	QueueOutgoingMessage<FCommandResponse>(RequestId, *Response);
 }
 
-void USpatialWorkerConnection::SendCommandFailure(Worker_RequestId RequestId, const char* Message)
+void USpatialWorkerConnection::SendCommandFailure(Worker_RequestId RequestId, const FString& Message)
 {
-	Worker_Connection_SendCommandFailure(WorkerConnection, RequestId, Message);
+	QueueOutgoingMessage<FCommandFailure>(RequestId, Message);
 }
 
-void USpatialWorkerConnection::SendLogMessage(const uint8_t Level, const char* LoggerName, const char* Message)
+void USpatialWorkerConnection::SendLogMessage(const uint8_t Level, const FName& LoggerName, const TCHAR* Message)
 {
-	Worker_LogMessage LogMessage{};
-	LogMessage.level = Level;
-	LogMessage.logger_name = LoggerName;
-	LogMessage.message = Message;
-
-	Worker_Connection_SendLogMessage(WorkerConnection, &LogMessage);
+	QueueOutgoingMessage<FLogMessage>(Level, LoggerName, Message);
 }
 
-void USpatialWorkerConnection::SendComponentInterest(Worker_EntityId EntityId, const TArray<Worker_InterestOverride>& ComponentInterest)
+void USpatialWorkerConnection::SendComponentInterest(Worker_EntityId EntityId, TArray<Worker_InterestOverride>&& ComponentInterest)
 {
-	Worker_Connection_SendComponentInterest(WorkerConnection, EntityId, ComponentInterest.GetData(), ComponentInterest.Num());
+	QueueOutgoingMessage<FComponentInterest>(EntityId, MoveTemp(ComponentInterest));
 }
 
 Worker_RequestId USpatialWorkerConnection::SendEntityQueryRequest(const Worker_EntityQuery* EntityQuery)
 {
-	return Worker_Connection_SendEntityQueryRequest(WorkerConnection, EntityQuery, 0);
+	QueueOutgoingMessage<FEntityQueryRequest>(*EntityQuery);
```

Comment:
We need to pass ownership of any `Worker_Constraint*` in `And`, `Or`, or `Not` constraints.

Reply:
As well as `snapshot_result_type_component_ids` if we're passing it.